magento/magento2
# Solved issue with applying shopping cart rule to a single bundle item

### Description (*)
There are two kinds of parent products: Bundle and Configurable (Grouped does not present in the quote)
For the Configurable product - we should calculate discount for the parent.
For the Bundle product - we should calculate children and ignore the Parent.

### Related Pull Requests

### Fixed Issues (if relevant)
1. Fixes magento/magento2#28921

### Manual testing scenarios (*)
1. Create a bundle product with two option each contain two product in it
2. Create one option as main as the title with two simple product (a&b) - (Dropdown)
3. Create the second option as a sub as the title with two simple product(c&d)- (Checkbox)
4. Create a shopping cart rule
5. Select a "percent of product price discount"
6. Add 10 % to the discount amount
7. We should add the main option two product in the rule like( SKU is one of (a&b)
8. Apply the rule
9. Go to the frontend
10. Select one SKU from the dropdown of the main simple product and checkbox one sub simple product
11. Add to cart
12. Go to cart page
13. Check the discount is applied total of two product

### Questions or comments
All assumptions are in the description section

### Contribution checklist (*)
 - [*] Pull request has a meaningful description of its purpose
 - [*] All commits are accompanied by meaningful commit messages
 - [*] All new or changed code is covered with unit/integration tests (if applicable)
 - [*] All automated tests passed successfully (all builds are green)
